tests/foreman/ui/test_host.py

```diff
@@ -2985,3 +2987,144 @@ def test_positive_manage_repository_sets(
         for content_host in content_hosts:
             output = content_host.execute('subscription-manager repos --list').stdout
             assert "Enabled:   1" in output, 'repository status not changed'
+
+
+def test_disassociate_multiple_hosts(
+    new_host_ui,
+    request,
+    target_sat,
+    module_location,
+    module_org,
+    vmware,
+    default_location,
+):
+    """
+    Import multiple VMs from a VMware compute resource, disassociate them via the UI,
+    and verify via API that their uuid and compute_resource_id are cleared.
+
+    :id: e5af21c7-62ef-4cc7-a72a-ab6c26090b68
+
+    :steps:
+        1. Create all required entities (domain, subnet, hostgroup, etc.)
+        2. Import 2 VMs from VMware into Satellite
+        3. Disassociate the VMs via the All Hosts UI
+        4. Verify via API that uuid and compute_resource_id are None
+
+    :expectedresults: VMs are disassociated and their compute resource info is cleared.
+
+    :CaseComponent: Hosts-Content
+
+    :Team: Phoenix-subscriptions
+    """
+
+    cr_name = gen_string('alpha')
+    hostgroup_name = gen_string('alpha')
+
+    # create entities for hostgroup
+    target_sat.api.SmartProxy(
+        id=target_sat.nailgun_smart_proxy.id, location=[default_location.id, module_location.id]
+    ).update()
+    domain = target_sat.api.Domain(
+        organization=[module_org.id], location=[module_location]
+    ).create()
+    subnet = target_sat.api.Subnet(
+        organization=[module_org.id], location=[module_location], domain=[domain]
+    ).create()
+    architecture = target_sat.api.Architecture().create()
+    ptable = target_sat.api.PartitionTable(
+        organization=[module_org.id], location=[module_location]
+    ).create()
+    operatingsystem = target_sat.api.OperatingSystem(
+        architecture=[architecture], ptable=[ptable]
+    ).create()
+    medium = target_sat.api.Media(
+        organization=[module_org.id], location=[module_location], operatingsystem=[operatingsystem]
+    ).create()
+    lce = (
+        target_sat.api.LifecycleEnvironment(name="Library", organization=module_org.id)
+        .search()[0]
+        .read()
+        .id
+    )
+    cv = target_sat.api.ContentView(organization=module_org).create()
+    cv.publish()
+
+    # create hostgroup
+    hostgroup_name = gen_string('alpha')
```

Comment:
This is the second definition of `hostgroup_name`. I would just remove the one from line 3021 and keep this one.